enclave/src/io.rs

```diff
@@ -0,0 +1,102 @@
+/*
+	Copyright 2019 Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+use std::fs::File;
+use std::io::{Read, Write};
+use std::sgxfs::SgxFile;
+use std::vec::Vec;
+
+use sgx_types::*;
+
+use log::*;
+
+pub fn write_file(bytes: &[u8], filepath: &str) -> SgxResult<sgx_status_t> {
+	match SgxFile::create(filepath) {
+		Ok(mut f) => match f.write_all(bytes) {
+			Ok(()) => {
+				info!("[Enclave] Writing keyfile '{}' successful", filepath);
+				Ok(sgx_status_t::SGX_SUCCESS)
+			}
+			Err(x) => {
+				error!("[Enclave -] Writing keyfile '{}' failed! Err: {}", filepath, x);
+				Err(sgx_status_t::SGX_ERROR_UNEXPECTED)
+			}
+		},
+		Err(x) => {
+			error!("[Enclave !] Creating keyfile '{}' Err: {}", filepath, x);
+			Err(sgx_status_t::SGX_ERROR_UNEXPECTED)
+		}
+	}
+}
+
+pub fn read_file(filepath: &str) -> SgxResult<Vec<u8>> {
+	let mut keyvec: Vec<u8> = Vec::new();
+	match SgxFile::open(filepath) {
+		Ok(mut f) => match f.read_to_end(&mut keyvec) {
+			Ok(len) => {
+				info!("[Enclave] Read {} bytes from key file", len);
+				Ok(keyvec)
+			}
+			Err(x) => {
+				error!("[Enclave] Read sealed file failed {}: Err {}", filepath, x);
+				Err(sgx_status_t::SGX_ERROR_UNEXPECTED)
+			}
+		},
+		Err(x) => {
+			info!("[Enclave] Can't find sealed file {} Err: {}" , filepath, x);
```

Comment:
Should by type `error!`

Reply:
it is only an error if we can't recover from it. However, in the context of this function it actually is an error. Maybe we could do `file_exists() -> Bool` before calling this function?

Reply:
The code is simplified now. But i still left the check inside the function as it is always needed, why not put it inside then?